### PR TITLE
Add a fixed-seed API to the simulator

### PR DESCRIPTION
**Pull request by josephlr**
This allows the simulator to behave deterministicly for the purposes of testing.

### CODE REVIEW

File: simulator/simulator.go [hunk 1]
```diff
@@ -73,6 +79,23 @@ func Get() (*Simulator, error) {
...
 }
 
+// GetWithFixedSeed behaves like Get() expect that all of the internal
+// heirarchy seeds are derived from the input seed.
+func GetWithFixedSeed(seed int) (*Simulator, error) {
+	s, err := Get()
+	if err != nil {
+		return nil, err
+	}
+	for i := 0; i < 64; i++ {
```
Comment (twitchy-jsonp) [inline simulator/simulator.go]: Is this intended to be deterministically puesdorandom? Could we just use `math/rand` if so?

Comment (josephlr) [reply simulator/simulator.go]: Fixed to use math.rand

File: simulator/simulator_test.go [hunk 2]
```diff
@@ -84,3 +84,36 @@ func TestGetRandom(t *testing.T) {
...
+func TestFixedSeedExpectedModulus(t *testing.T) {
+	s, err := GetWithFixedSeedInsecure(0)
+	if err != nil {
+		t.Fatal(err)
+	}
+	defer s.Close()
+
+	modulus := getEKModulus(t, s)
+	if modulus.Cmp(zeroSeedModulus()) != 0 {
+		t.Fatalf("Got %v expected %v", modulus, zeroSeedModulus())
```
Comment (twitchy-jsonp) [inline simulator/simulator_test.go]: I think asserts are meant to have error messages that stand alone (though not 100% on this one).

Thoughts on: `getEKModulus() = %v, want %v` ?

Comment (josephlr) [reply simulator/simulator_test.go]: Good idea, changed

**Pull request merged**